mlir/lib/Dialect/Vector/Transforms/VectorTransforms.cpp

```diff
@@ -1607,7 +1607,27 @@ struct ChainedReduction final : OpRewritePattern<vector::ReductionOp> {
   }
 };
 
-/// For vectors with either leading or trailing unit dim, replaces:
+/// Drop unit dimension of the input VectorType. Scalable dimensions cannot be
+/// folded as we do not want to merge them through a shape_cast and create ill
+/// shaped scalable sizes.
```

Comment:
>  Scalable dimensions cannot be folded as we do not want to merge them through a shape_cast and create ill shaped scalable sizes.

This is a bit more nuanced :)

> Scalable unit dimensions are not supported. Folding such dimensions would require "shifting" the scalable flag onto some other fixed-width dim (e.g. `vector<[1]x4xf32>` -> `vector<[4]xf32>`). This could be implemented in the future.

@MacDue Was there any other issue with unit scalable dims that I missed here?

Reply:
Not really, the main point is you have have to preserve the number of elements in the vector type. A unit dim does not change the number of elements in a vector (so can be dropped). A scalable one dim (which is not a unit dim) contains vscale elements, so dropping it divides the number of elements in the vector by vscale -- which obviously does not preserve the number of elements in the vector type.

Reply:
@nujaa Are you OK to update this comment as per my suggestion above? Thanks!

Reply:
Hi, done. sorry, I must have misread the comment last week.